doc/aaa/AAA Restrictions/AAA Restrictions.md

```diff
@@ -0,0 +1,252 @@
+
+### Definitions/Abbreviations 
+- **API**: Application Programmable Interface
+- **AAA**: Authentication, Authorization, and Accounting
+- **PAM**: Pluggable Authentication Module
+
+
+### Reference
+|  Ref link                                                                                            |  Description                                              |  
+| :--------------------------------------------------------------------------------------------------: | :-------------------------------------------------------: |
+|  https://github.com/sonic-net/SONiC/blob/master/doc/aaa/AAA%20Improvements/AAA%20Improvements.md     | Sonic AAA improvements                                    |
+|  https://man7.org/linux/man-pages/man3/pam.3.html                                                    | Linux PAM                                                 |
+|  https://confluence.nvidia.com/pages/viewpage.action?pageId=1347495920                               | Pluggable Authentication Modules Library                  |
+|  http://pubs.opengroup.org/onlinepubs/8329799/chap5.htm                                              | Implementation details of ldap pam & nss                  |
+|  http://wpollock.com/AUnix2/PAM-Help.htm                                                             | PAM options                                               |
+|  https://packages.debian.org/sid/libpam-ldapd                                                        | PAM options                                               | 
+
+### Overview 
+AAA (Authentication, Authorization, and Accounting) in Sonic establishes a robust framework for access control, policy enforcement, usage auditing, and service billing, supporting standard protocols such as RADIUS, TACACS+, and LDAP for user authentication.
+
+This HLD emphasizes enhancing the security of aaa by introducing "restrictions" on the number and frequency of login attempts. The focus is on augmenting the system's security by preventing brute force attacks and ensuring that user authentication attempts are both monitored and controlled effectively.
+
+### Requirements
+
+-      Limit Login Attempts: Specify the maximum number of unsuccessful login attempts allowed before a user account is temporarily locked. This prevents continuous guessing of credentials.
+-      Control Login Speed: Introduce delays between consecutive login attempts to slow down any automated login attempts, significantly reducing the risk of brute force attacks.
+-      Define Unlock Time: Establish a configurable unlock time that specifies how long a user account remains locked before allowing another login attempt. This control helps manage the access of users who have shown suspicious login behaviors.
+-      TODO: SONIC rest API/nginx?
+
+### Architecture Design 
+
+Arc design diagram\
+![arc_aaa_sonic](restrictions.png)
```

Comment:
Link does not work

Reply:
when you open it using the branch in preview it shows the image (for me at least)